src/main/scala/org/dia/algorithms/mcc/MCCEdge.scala

```diff
@@ -19,21 +19,29 @@ package org.dia.algorithms.mcc
 
 class MCCEdge(var srcNode : MCCNode, var destNode: MCCNode, var weight : Double) extends Serializable {
 
+  var areaOverlap: Double = 0.0
+
+  def incrementAreaOverlap(): Unit = {
+    areaOverlap += 1
+  }
+
+  def updateWeight(_weight: Double): Unit = {
+    weight = _weight
```

Comment:
@sujen1412 in keeping with the style of the codebase, please make these changes accordingly.